sphinx_doc_src/faq.rst

```diff
@@ -112,6 +112,28 @@ You can also declare a local variable in a block::
     end
 
 
+How do I check whether a variable is defined?
+---------------------------------------------
+
+Use ``set -q var``.  For example, ``if set -q var; echo variable defined; end``.  To check multiple variables you can combine with ``and`` and ``or`` like so::
+
+    if set -q var1; or set -q var2
+        echo either variable defined
+    end
+
+Keep in mind that a defined variabled could be empty. Both of these commands define an empty variable: ``set var`` and ``set var ""``.
```

Comment:
These are still different, though!

```fish
$ set var
$ count $var
0
$ set var ''
$ count $var
1
```

But the underlying question there is, of course, how deep we want to go into it for a FAQ entry.

Do we want to go into `$var` vs `$var[1]`, do we want to add `set -ql`? Do we mention `test`?

See also [one of my more popular stackoverflow answers](https://stackoverflow.com/questions/47743015/fish-shell-how-to-check-if-a-variable-is-set-empty/47743269#47743269).